Lib/email/mime/audio.py

```diff
@@ -36,65 +35,20 @@ def __init__(self, _audiodata, _subtype=None,
         constructor, which turns them into parameters on the Content-Type
         header.
         """
-        if _subtype is None:
-            _subtype = _what(_audiodata)
-        if _subtype is None:
-            raise TypeError('Could not find audio MIME subtype')
+        _subtype = _subtype or _infer_subtype(_audiodata)
```

Comment:
This changes the behavior if _subtype is '' (or any other non-None null value).   Someone other than me needs to agree with this.

Reply:
I personally prefer more explicit tests than just "falseiness" when a more explicit type is indicated.  In this case `_subtype=None` defaults to None to mean that this argument wasn't given.